src/Runner.Worker/ActionManager.cs

```diff
@@ -907,7 +915,14 @@ private async Task DownloadRepositoryActionAsync(IExecutionContext executionCont
                 Directory.CreateDirectory(stagingDirectory);
 
 #if OS_WINDOWS
-                ZipFile.ExtractToDirectory(archiveFile, stagingDirectory);
+                try
+                {
+                    ZipFile.ExtractToDirectory(archiveFile, stagingDirectory);
+                }
+                catch (InvalidDataException e)
+                {
+                    throw new InvalidActionArchiveException($"Can't un-zip archive file: {archiveFile}. action being checked out: {downloadInfo.NameWithOwner}. error: {e}.");
```

Comment:
Can we do `{downloadInfo.NameWithOwner}@{downloadInfo.Ref}`

The backend url is based on both Name and ref so this would help us correlate better
```suggestion
                    throw new InvalidActionArchiveException($"Can't un-zip archive file: {archiveFile}. action being checked out: {downloadInfo.NameWithOwner}@{downloadInfo.Ref}. error: {e}.");
```